libs/workspace/src/schematics/dt-component-e2e/index.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DtComponentE2EOptions } from './schema';
+import { strings } from '@angular-devkit/core';
+import { join } from 'path';
+import {
+  Rule,
+  Tree,
+  chain,
+  apply,
+  url,
+  template,
+  mergeWith,
+} from '@angular-devkit/schematics';
+import * as ts from 'typescript';
+import { getSourceFile, findNodes, getIndentation } from '../utils/ast-utils';
+import { InsertChange, commitChanges } from '../utils/change';
+
+interface DtE2EExtendedOptions {
+  selector: string;
+  componentModule: {
+    name: string;
+    package: string;
+  };
+  e2eComponent: {
+    component: string;
+    module: string;
+  };
+  name: string;
+}
+
+function generateComponentOptions(
+  name: string,
+): { name: string; package: string } {
+  return {
+    name: `Dt${strings.classify(name)}Module`,
+    package: `@dynatrace/barista-components/${strings.dasherize(name)}`,
+  };
+}
+
+function generateE2EComponentOptions(
+  name: string,
+): { component: string; module: string } {
+  return {
+    component: `DtE2E${strings.classify(name)}`,
+    module: `DtE2E${strings.classify(name)}Module`,
+  };
+}
+
+/**
+ * Adds a new route inside the ui-test routes
+ */
+export function addRoute(options: DtE2EExtendedOptions): Rule {
+  return (host: Tree) => {
+    const modulePath = join(
+      'apps',
+      'components-e2e',
+      'src',
+      'app',
+      'app.routing.module.ts',
+    );
```

Comment:
No need for joining here: If you write the path as a string you gain multiple benefits in this case. 
1. It gets way more readable 
2. If you need to adapt paths and you utilize the search functionality and you search for `apps/components-e2e/` then you will find this occurrence otherwise you may forget about it.

```suggestion
    const modulePath = 'apps/components-e2e/src/app/app.routing.module.ts';
```

Reply:
Not joining might result in windows path issues - please keep the join!